src/render/draw_hillshade.js

```diff
@@ -114,11 +121,18 @@ function prepareHillshade(painter, tile, layer, depthMode, stencilMode, colorMod
     context.bindFramebuffer.set(fbo.framebuffer);
     context.viewport.set([0, 0, tileSize, tileSize]);
 
+    let tileBoundsBuffer, tileBoundsIndexBuffer, tileBoundsSegments;
+    if (tile.tileBoundsSegments) {
+        ({tileBoundsBuffer, tileBoundsIndexBuffer, tileBoundsSegments} = tile);
+    } else {
+        ({tileBoundsBuffer, quadTriangleIndexBuffer: tileBoundsIndexBuffer, tileBoundsSegments} = painter);
+    }
```

Comment:
This chunk of code is repeated many times across different files so it should be refactored into a method, e.g. `painter.getTileBounds(tile)`.